src/Service/CartDuplicationService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Invertus\SaferPay\Service;
+
+use Cart;
+use CartRule;
+use Context;
+use Db;
+use Order;
+
+class CartDuplicationService
+{
+    public function restoreCart($cartId)
```

Comment:
we might face a case where we loose products from cart  - we should handle this case by displaying an error message atleast.